paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/orphan/SparkOrphanFilesClean.java

```diff
@@ -0,0 +1,250 @@
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import scala.Tuple2;
+
+import static org.apache.paimon.utils.Preconditions.checkArgument;
+
+/** Spark {@link OrphanFilesClean}, it will submit a job for a table. */
+public class SparkOrphanFilesClean extends OrphanFilesClean {
+
+    @Nullable protected final Integer parallelism;
+
+    public SparkOrphanFilesClean(
+            FileStoreTable table,
+            long olderThanMillis,
+            SerializableConsumer<Path> fileCleaner,
+            @Nullable Integer parallelism) {
+        super(table, olderThanMillis, fileCleaner);
+        this.parallelism = parallelism;
+    }
+
+    public static long executeDatabaseOrphanFiles(
+            JavaSparkContext ctx,
+            Catalog catalog,
+            long olderThanMillis,
+            SerializableConsumer<Path> fileCleaner,
+            @Nullable Integer parallelism,
+            String databaseName,
+            @Nullable String tableName)
+            throws Catalog.DatabaseNotExistException, Catalog.TableNotExistException {
+        Long cleanTotal = 0L;
+        List<String> tableNames = Collections.singletonList(tableName);
+        if (tableName == null || "*".equals(tableName)) {
+            tableNames = catalog.listTables(databaseName);
+        }
+
+        for (String t : tableNames) {
+            Identifier identifier = new Identifier(databaseName, t);
+            Table table = catalog.getTable(identifier);
+            checkArgument(
+                    table instanceof FileStoreTable,
+                    "Only FileStoreTable supports remove-orphan-files action. The table type is '%s'.",
+                    table.getClass().getName());
+
+            Long cleanNums =
+                    new SparkOrphanFilesClean(
+                                    (FileStoreTable) table,
+                                    olderThanMillis,
+                                    fileCleaner,
+                                    parallelism)
+                            .doOrphanClean(ctx);
+            cleanTotal += cleanNums;
+        }
+        return cleanTotal;
+    }
+
+    @Nullable
+    public Long doOrphanClean(JavaSparkContext ctx) {
+        int parallelism = this.parallelism == null ? ctx.defaultParallelism() : this.parallelism;
+        ctx.getConf().set("spark.default.parallelism", String.valueOf(parallelism));
+        SparkContext sc = ctx.sc();
+
+        List<String> branches = validBranches();
+
+        // snapshot and changelog files are the root of everything, so they are handled specially
+        // here, and subsequently, we will not count their orphan files.
+        AtomicLong deletedInLocal = new AtomicLong(0);
+        cleanSnapshotDir(branches, p -> deletedInLocal.incrementAndGet());
+
+        // branch and manifest file
+        CollectionAccumulator<Pair<String, String>> pairsAccumulator =
```

Comment:
Why we need a `CollectionAccumulator`? Why not just use RDD?

Reply:
Thanks @JingsongLi       I consider 2 points:
1. Want to reuse OrphanFilesClean##collectWithoutDataFile method and Spark API seems not contain side output stream like Flink  
2. If use rdd maybe need split another rdd. CollectionAccumulator work as side output.

Reply:
Here CollectionAccumulator change to Rdd is better, had changed it and do some change in parent method for more common calling. @JingsongLi